neuralmonkey/decoders/autoregressive.py

```diff
@@ -119,6 +143,14 @@ def batch_size(self) -> tf.Tensor:
 
     @tensor
     def decoding_w(self) -> tf.Variable:
+        if (self.tie_embeddings
+                and self.embedding_size != self.output_dimension):
+            raise ValueError(
+                "`embedding_size must be equal to the output_projection "
+                "size when using the `tie_embeddings` option")
```

Comment:
tadyta kontrola by měla bejt asi už v konstruktoru. plus nevim co se stane když budeš porovnávat `int` a `tf.Tensor`.

Reply:
(`self.output_dimension` může bejt `tf.Tensor`)

Reply:
možná bych tu kontrolu vyhodil

Reply:
V konstruktoru to byt nemuze, protoze AutoregressiveDecoder.output_dimension je abstract_property, ktera je definovana az v potomkovi, ktery se inicializuje az po predkovi.

Reply:
Klidne bych to vyhodil, ale tu kontrolu jsem tam dal na podnet @jlibovicky

Reply:
No tak jak je to teď je to zkrátka špatně, protože `3 != tf.constant(3)` a `output_dimension` může vracet tenzor.

Reply:
A musi vubec output_dimension byt tf.Tensor? Kdyz se koukam do zdrojaku, tak vzdycky tam je int.

Reply:
to je potřeba ověřit. Ale záleží to na tom, co použiješ za output projection u dekodéru. některý output projekce to podle mě staticky mít nemusí.

Reply:
Takze:
transformer.output_dimension je int primo z definice
decoder.output_dimension je budto decoder.rnn_size (int) a nebo decoder.output_projection_spec[1] (opet int)

Reply:
Btw, nemel by to nahodou mypy detekovat, pokud by to tak nebylo?

Reply:
mypy bere `tf.Tensor` jako `Any`, takže ti tady nic nenahlásí